compiler/ballerina-lang/src/main/resources/grammar/BallerinaLexer.g4

```diff
@@ -341,7 +341,7 @@ EscapeSequence
 
 fragment
 UnicodeEscape
-    :   '\\' 'u' HexDigit HexDigit HexDigit HexDigit
+    :   '\\' 'u' LEFT_BRACE HexDigit+ RIGHT_BRACE
```

Comment:
Should we change HexDigit* here to avoid a syntax error?

Reply:
According to the spec, Unicode is defined as,
```
StringNumericEscape := \u{ CodePoint }
CodePoint := HexDigit+
```